app/clients/sms/twilio.py

```diff
@@ -191,13 +191,13 @@ def send_sms(
             return message.sid
         except TwilioRestException as e:
             if e.status == 400 and 'phone number' in e.msg:
-                self.logger.error('Twilio send SMS request for %s failed, TwilioRestException - %s', reference, e.msg)
-                raise InvalidProviderException
+                self.logger.exception('Twilio send SMS request for %s failed', reference)
+                raise InvalidProviderException from e
```

Comment:
What is the correct syntax? 

I see in the file `app/attachments/store.py` on line 87 the syntax is as follows `raise AttachmentStoreError() from e`. Above `InvalidProviderException` does not have `()`. Which syntax is correct?

Reply:
Either works.
```python3
>>> try:
...     raise RuntimeError
... except RuntimeError as e:
...     raise ValueError from e
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
RuntimeError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ValueError
```
```python
>>> try:
...     raise RuntimeError
... except RuntimeError as e:
...     raise ValueError() from e
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
RuntimeError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ValueError
```

Reply:
Good to know